tests/jdom/button.spec.ts

```diff
@@ -0,0 +1,136 @@
+// Test the ButtonServer, intended as a test for the unit test framework.
+// Yes, we're meta-testing!
+
+import { suite, test } from "mocha"
+import { ButtonEvent } from "../../src/jdom/constants"
+import { withBus, createServices, nextEventFrom } from "./tester"
+import { assert } from "../../src/jdom/utils"
+import ButtonServer from "../../src/servers/buttonserver"
+
+suite("button server", () => {
+    test("fires edge events after changing state", async function () {
+        await withBus(async bus => {
+            const { button } = await createServices(bus, {
+                button: new ButtonServer("button", false),
+            })
+
+            button.server.down()
+            assert(
+                (await nextEventFrom(button.service, { within: 100 })).code ==
+                    ButtonEvent.Down
+            )
+
+            button.server.up()
+            assert(
+                (await nextEventFrom(button.service, { within: 100 })).code ==
+                    ButtonEvent.Up
+            )
+        })
+    })
+
+    test("fires both down and hold events when held", async function () {
+        await withBus(async bus => {
+            const { button } = await createServices(bus, {
+                button: new ButtonServer("button", false),
+            })
+
+            button.server.down()
+            assert(
+                (await nextEventFrom(button.service, { within: 100 })).code ==
+                    ButtonEvent.Down
+            )
+            assert(
+                (
+                    await nextEventFrom(button.service, {
+                        after: 500,
+                        tolerance: 100,
+                    })
+                ).code == ButtonEvent.Hold
+            )
+        })
+    })
+
+    test("repeatedly raise hold events when held", async function () {
```

Comment:
we can save a line :)
```
    test("repeatedly raise hold events when held", withBus(async bus => {
```

Reply:
That does not work, `test(...)` requires a function (instead of the returned value of a function).

It is possible to use the more concise `() => ...` function syntax, but prettier still forces a new line there. And if we have `withBus` return a function, prettier puts each argument to `test` on its own line.